src/main/java/io/kestra/plugin/influxdb/FluxQuery.java

```diff
@@ -0,0 +1,295 @@
+        title = "Flux query to execute",
+        description = "The Flux query to run against InfluxDB"
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    private Property<String> query;
+
+    @Schema(
+        title = "Whether to store the data from the query result into an ion serialized data file."
+    )
+    @Builder.Default
+    private Property<Boolean> store = Property.of(false);
+
+    @Schema(
+        title = "The way you want to store the data.",
+        description = "FETCH_ONE output the first row, "
+            + "FETCH output all the rows, "
+            + "STORE store all rows in a file, "
+            + "NONE do nothing."
+    )
+    @PluginProperty
+    @Builder.Default
+    private Property<FetchType> fetchType = Property.of(FetchType.FETCH);
+
+    public enum FetchType {
+        FETCH,
+        FETCH_ONE,
+        STORE,
+        NONE
+    }
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        Logger logger = runContext.logger();
+        String renderedQuery = runContext.render(query).as(String.class).orElseThrow();
+
+        try (
+            InfluxDBClient client = this.connection.client(runContext);
+        ){
+            QueryApi queryApi = client.getQueryApi();
+            logger.debug("Starting query: {}", query);
+
+            List<FluxTable> tables = queryApi.query(renderedQuery);
+
+            Output.OutputBuilder outputBuilder = Output.builder();
+            int recordCount = 0;
+
+            for (FluxTable table : tables) {
+                recordCount += table.getRecords().size();
+            }
+
+            switch (runContext.render(fetchType).as(FetchType.class).orElseThrow()) {
+                case FETCH:
+                    Pair<List<Map<String, Object>>, Integer> fetch = this.fetch(tables);
+                    outputBuilder
+                        .rows(fetch.getLeft())
+                        .size(fetch.getRight());
+                    break;
+
+                case FETCH_ONE:
+                    Map<String, Object> row = this.fetchOne(tables);
+                    outputBuilder
+                        .row(row)
+                        .size(row != null ? 1 : 0);
+                    break;
+
+                case STORE:
+                    Pair<URI, Long> store = this.store(runContext, tables);
+                    outputBuilder
+                        .uri(store.getLeft())
+                        .size(store.getRight().intValue());
+                    break;
+
+                case NONE:
+                    outputBuilder.size(recordCount);
+                    break;
+            }
+
+            runContext.metric(Counter.of("records", recordCount));
+            return outputBuilder.total((long) tables.size()).build();
+        }
+    }
+
+    protected Pair<List<Map<String, Object>>, Integer> fetch(List<FluxTable> tables) {
+        List<Map<String, Object>> results = new ArrayList<>();
+        int count = 0;
```

Comment:
The count seems to be the same as the size of the results list.
If so, it can be avoided and you can use the size of the list instead.